vars/mavenExecute.groovy

```diff
@@ -55,13 +59,20 @@ private void validateParameter(parameter, name) {
 
     for (int i = 0; i < parameter.size(); i++) {
         String element = parameter[i]
+        validateStringParameter(element)
+    }
+}
 
-        if (element =~ /-D.*='.*'/) {
-            echo "[$STEP_NAME WARNING] It looks like you passed a define in the form -Dmy.key='this is my value' in $element. Please note that the quotes might cause issues. Correct form: -Dmy.key=this is my value"
-        }
+private void validateStringParameter(String element) {
+    if (!element) {
+        return
+    }
 
-        if (element.startsWith("'") && element.endsWith("'") && element.contains(' ')) {
-            echo "[$STEP_NAME WARNING] It looks like $element is quoted but it should not be quoted."
-        }
+    if (element =~ /-D.*='.*'/) {
+        echo "[$STEP_NAME WARNING] It looks like you passed a define in the form -Dmy.key='this is my value' in $element. Please note that the quotes might cause issues. Correct form: -Dmy.key=this is my value"
+    }
+
+    if (element.startsWith("'") && element.endsWith("'") && element.contains(' ')) {
```

Comment:
I think the quotes are wrong regardless. And they may have been added unconditionally, which the third condition would prevent from catching.

```suggestion
    if (element.length() >= 2 && element.startsWith("'") && element.endsWith("'")) {
```